File: KalimaSmartContracts/sensors.js
/**
	Auteur : Jérôme Delaire
	Date : 09/03/2021
	Script de démo pour créer une transaction sur Kalima
**/
load("nashorn:mozilla_compat.js");
importPackage(Packages.java.io);
importPackage(Packages.java.lang);
importPackage(Packages.java.util);
importPackage(org.kalima.kalimamq.message);
importPackage(org.kalima.cache.lib);
importPackage(org.kalima.util);

var JString = Java.type("java.lang.String");

function main(kMsg, clone, logger) {
	
	if(!kMsg.getKey().equals("temperature2")) return;
	
	var body = new JString(kMsg.getBody());
	var temperature = parseInt(body, 10);
	if(temperature == null)
		return "NOK";

	if(temperature >= 100) {
		clone.put("/alarms/fire", kMsg.getKey(), ("Temperature too high: " + temperature + " °C").getBytes());
	}
}

File: KalimaCSmartContract/src/LuaScripts/sensors.lua
function functions(List)
    add_function(List,"LuaGetBody")
    add_function(List,"LuaGetKey")
    add_function(List,"LuaPutMsg")
    add_function(List,"LuaPutLog")
    add_function(List,"LuaStrlen")
end

function test(Kmsg,Node)
    key = LuaGetKey(Kmsg)
    if(key=="temperature")
    then
        body = LuaGetBody(Kmsg)
        LuaPutLog(Node,"body : " .. body)
        if(tonumber(body)>=100)
        then
            LuaPutMsg(Node,"/alarms/fire", 12, "temperature is too high", 23, body, LuaStrlen(body))
        end
    end
end